docs/source/testing-and-debugging.md

```diff
@@ -43,71 +43,79 @@ from boa3.boa3 import Boa3
+error will be raised.
 
-You'll have to call the method `call_contract()` to interact with your smart contract. Its parameters are the path of
-the compiled smart contract, the smart contract's method, and the arguments if necessary. 
-This call doesn't return the result directly, but includes it in a queue of invocations. To execute all the invocations
-set up, call the method `execute()`. Then assert the result of your invoke to see if it's correct.
-
-Note that `invoke.result` won't be set if the execution fails, so you should also assert if `runner.vm_state` is valid 
-for your test case.
+To persist an invocation, use the `signing_accounts` parameter to pass a list of signing accounts when calling the 
+smart contract. If you don't pass it, then it will always be a test invoke. The `signers` parameter can be used 
+alongside the `signing_accounts` if you want to change the witness scope of the invocation, or by itself if you want to
+test invoke but also define the signers of the transaction.
 
 Your Python Script should look something like this:
 
 ```python
-
-from boa3.sc.types import VMState
-from boa3_test.test_drive.testrunner.neo_test_runner import NeoTestRunner
-
-
-def test_hello_world_main():
-    neoxp_config_file = '{path-to-neo-express-config-file}'
-    project_root_folder = '{path-to-project-root-folder}'
-    path = f'{project_root_folder}/boa3_test/examples/hello_world.nef'
-    runner = NeoTestRunner(neoxp_config_file)
-
-    invoke = runner.call_contract(path, 'main')
-    runner.execute()
-    assert runner.vm_state is VMState.HALT
-    assert invoke.result is None
+import asyncio
+from boaconstructor import SmartContractTestCase
+from neo3.core import types
+from neo3.wallet.account import Account
+from neo3.contracts.contract import CONTRACT_HASHES
+
+GAS = CONTRACT_HASHES.GAS_TOKEN
+
+class SmartContractTest(SmartContractTestCase):
+    genesis: Account
+    user1: Account
+    
+    # if this variable is set, then this contract hash will be called whenever you don't use the `target_contract` parameter on the `call` method
+    contract_hash: types.UInt160
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        # you can name the account whatever you want, but the password needs to be "123"
+        cls.user1 = cls.node.wallet.account_new("123", "alice")
+        cls.genesis = cls.node.wallet.account_get_by_label("committee")
+
+        asyncio.run(cls.asyncSetupClass())
+
+    @classmethod
+    async def asyncSetupClass(cls) -> None:
+        # this `transfer` method already uses the correct amount of decimals for the token
+        await cls.transfer(GAS, cls.genesis.script_hash, cls.user1.script_hash, 100)
+        
+        # the smart contract I'm deploying is hello_world_with_deploy.py from the "Neo Methods" https://dojo.coz.io/neo3/boa/getting-started.html#neo-methods 
+        cls.contract_hash = await cls.deploy("./smart_contract.nef", cls.genesis)
```

Comment:
I think these code snippets should be along the texts that explain what to do, the text by itself does not help much without code to look into
If the user if following the documentation I don't think its test is going to look like this as the text before it suggests, there is mention to initiating accounts, but is not explicit that in this tutorial we're setting a user test account.